### DISCUSSION TITLE
Allow us to disable items from the config files

### DISCUSSION THREAD

**Discussion by purejosh** [TechReborn/TechReborn#3016] [Ideas]
Creating a datapack match to make certain items uncraftable is a hackabout fix. and there are many mods that provide built-in disables for many types of items, enchantments, etc. 

I personally find all of the armor and tools for this mod frivolous, and I downloaded it to focus on the energy aspect. Having a config file to disable certain subsets of items would be extremely helpful.

**Comment by Ayutac** [comment]
Isn't it normal to have a bit of item overhead in some mods? And datapacks are the way to go for modpacks, not a hackabout fix.

I feel like this is a bad idea, because then maybe everyone wants their own subset and us to make it possible. I guess technically we could omit certain items from being registered, but then the recipes would need to be removed too or we might get errors and all in all this is a can of worms.

If you don't want to use something from a mod, simply don't use it?

**Reply by purejosh** [reply]
> Isn't it normal to have a bit of item overhead in some mods? And datapacks are the way to go for modpacks, not a hackabout fix.
> 
> I feel like this is a bad idea, because then maybe everyone wants their own subset and us to make it possible. I guess technically we could omit certain items from being registered, but then the recipes would need to be removed too or we might get errors and all in all this is a can of worms.
> 
> If you don't want to use something from a mod, simply don't use it?

It's a bad idea to allow people the freedom to disable parts of mods that they're not fond of? That's quite possibly one of the silliest things I've heard. As a server owner, you should have control over anything that you feel is unbalanced.

**Reply by Ayutac** [reply]
as a modpack creator, you use datapacks. I think you underestimate the amount of work and potential bugs this degree of freedom in the code can cause. It doesn't seem to be worth the hassle when you can already archieve this via datapacks.

**Reply by purejosh** [reply]
I've seen many implementations of a disabling system that were flawless. I think the developer should consider it. Simple as that.

**Reply by Ayutac** [reply]
Do you have a reference mod? I would like to see how they solved it

**Reply by purejosh** [reply]
https://www.curseforge.com/minecraft/mc-mods/mcdw

**Reply by Technici4n** [reply]
It's not reasonable to ask every modder to make all of a mod disableable. You should be using KubeJS to mass-remove recipes for items that you don't want to use, and you can also use KubeJS to remove these items from REI's listing.
